File: mobile/lib/features/categories/domain/category.dart
class Category {
  const Category({required this.id, required this.name, this.description});

  final int id;
  final String name;
  final String? description;

  factory Category.fromJson(Map<String, dynamic> json) => Category(
        id: json['id'] as int,
        name: json['name'] as String,
        description: json['description'] as String?,
      );
}

File: backend/src/Servit.Domain/Entities/Category.cs
namespace Servit.Domain.Entities;

public class Category
{
    public int Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }

    public ICollection<ProviderCategory> ProviderCategories { get; set; } = new List<ProviderCategory>();
}
